src/silicone/database_crunchers/rms_closest.py

```diff
@@ -64,12 +76,21 @@ def derive_relationship(self, variable_follower, variable_leaders):
         iamdf_follower = self._get_iamdf_section(variable_follower)
         data_follower_time_col = iamdf_follower.time_col
         iamdf_lead = self._db.filter(variable=variable_leaders)
+        if not weighting:
+            weighting = {variab: 1 for variab in variable_leaders}
+        if any(var not in weighting.keys() for var in variable_leaders):
+            raise ValueError("Weighting does not include all lead variables.")
         iamdf_lead, iamdf_follower = _filter_for_overlap(
-            iamdf_lead, iamdf_follower, ["scenario", "model", data_follower_time_col]
+            iamdf_lead,
+            iamdf_follower,
+            ["scenario", "model", data_follower_time_col],
+            variable_leaders,
         )
 
-        leader_unit = _get_unit_of_variable(iamdf_lead, variable_leaders)
-        leader_unit = leader_unit[0]
+        leader_var_unit = {
+            var[1]["variable"]: var[1]["unit"]
+            for var in iamdf_lead.variables(True).iterrows()
```

Comment:
```suggestion
            for _, var in iamdf_lead.variables(True).iterrows()
```

and then tweaking the line above to just be `var`, rather than `var[1]` might be cleaner?

Reply:
yep, much better